src/prototype_creep_resources.js

```diff
@@ -1,9 +1,20 @@
 'use strict';
 
 Creep.prototype.harvesterBeforeStorage = function() {
-  var methods = [
-    Creep.getEnergy
-  ];
+  let methods = [];
+
+  if (this.memory.hasEnergy === undefined) {
+    this.memory.hasEnergy = (this.carry.energy == this.carryCapacity);
+  } else if (this.memory.hasEnergy && this.carry.energy === 0) {
+    this.memory.hasEnergy = false;
+  } else if (!this.memory.hasEnergy &&
+      this.carry.energy == this.carryCapacity) {
+    this.memory.hasEnergy = true;
+  }
```

Comment:
Source harvesting is done in the `getEnergy` method. If the creep hasEnergy the getEnergy method will not be executed. So the creep will harvest one time, has energy, next tick, getEnergy is not executed, the creep will move away.